FrontendVersion.compareBuildIdentifier throws NumberFormatException

FrontendVersion.compareBuildIdentifier throws NumberFormatException when comparing version "3.9999.0-SNAPSHOT".

```
Caused by: java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:592)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.vaadin.flow.server.frontend.FrontendVersion.compareBuildIdentifier(FrontendVersion.java:341)
	at com.vaadin.flow.server.frontend.FrontendVersion.compareTo(FrontendVersion.java:318)
	at com.vaadin.flow.server.frontend.FrontendVersion.isEqualTo(FrontendVersion.java:270)
	at com.vaadin.flow.server.frontend.NodeUpdater.handleExistingVaadinDep(NodeUpdater.java:404)
	at com.vaadin.flow.server.frontend.NodeUpdater.addDependency(NodeUpdater.java:382)
	at com.vaadin.flow.server.frontend.TaskUpdatePackages.updatePackageJsonDependencies(TaskUpdatePackages.java:182)
	at com.vaadin.flow.server.frontend.TaskUpdatePackages.execute(TaskUpdatePackages.java:116)
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:529)
	at com.vaadin.flow.plugin.maven.BuildFrontendMojo.runNodeUpdater(BuildFrontendMojo.java:204)
	at com.vaadin.flow.plugin.maven.BuildFrontendMojo.execute(BuildFrontendM
```

We're using an npm version "3.9999.0-SNAPSHOT" and that one is causing trouble right now. After a full cleanup this occurs after _second_ build when comparing current web component versions with the ones from package.json (is my guess).

Running following unit test

with flow-server-2.6.5 (V14.6.5) works
but with flow-server-2.7.1 (V14.7.1) it doesn't


```
    @Test
    public void testFrontendVersion() {
        FrontendVersion f1 = new FrontendVersion("3.9999.0-SNAPSHOT");
        FrontendVersion f2 = new FrontendVersion("3.9999.0-SNAPSHOT");

        f1.compareTo(f2);
    }
```

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendVersion.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/FrontendVersion.java
@@ -338,6 +338,12 @@ public class FrontendVersion
                 return thisMatcher.group(1)
                         .compareToIgnoreCase(otherMatcher.group(1));
             }
+            // if one or both are missing numeric value do not parse int
+            if (thisMatcher.group(2).isEmpty()
+                    || otherMatcher.group(2).isEmpty()) {
+                return buildIdentifier
+                        .compareToIgnoreCase(other.buildIdentifier);
+            }
             return Integer.parseInt(thisMatcher.group(2))
                     - Integer.parseInt(otherMatcher.group(2));
         }

--- a/flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendVersionTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendVersionTest.java
@@ -3,6 +3,7 @@ package com.vaadin.flow.server.frontend;
 import org.junit.Assert;
 import org.junit.Test;
 
+import static com.helger.commons.mock.CommonsAssert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
 
@@ -75,6 +76,18 @@ public class FrontendVersionTest {
         fromConstructor = new FrontendVersion(12, 3, 5, "alpha12");
         assertTrue("Full version with build identifier didn't match",
                 fromString.isEqualTo(fromConstructor));
+    }
+
+    @Test // #12041
+    public void testSimilarBuildIdentifiers() {
+        FrontendVersion version = new FrontendVersion("1.1.1-SNAPSHOT");
+        FrontendVersion equals = new FrontendVersion("1.1.1-SNAPSHOT");
+
+        assertTrue("Versions be the same", version.isEqualTo(equals));
+        assertFalse("Version should not be older", version.isOlderThan(equals));
+        assertEquals("Versions should not have a difference", 0,
+                version.compareTo(equals));
+        assertFalse("Version should not be newer", version.isNewerThan(equals));
     }
 
     @Test(expected = NumberFormatException.class)

